pbrd/pbr_map.c

```diff
@@ -101,6 +103,24 @@ static bool pbrms_is_installed(const struct pbr_map_sequence *pbrms,
 	return false;
 }
 
+void pbr_set_match_clause_for_pcp(struct pbr_map_sequence *pbrms, uint8_t pcp)
+{
+	if (pbrms) {
+		pbrms->match_pcp = pcp;
+		pbr_map_check(pbrms, true);
+	}
+}
+
+void pbr_set_match_clause_for_vlan(struct pbr_map_sequence *pbrms,
+				   uint16_t vlan_id, uint16_t vlan_flags)
+{
+	if (pbrms) {
+		pbrms->match_vlan_id = vlan_id;
+		pbrms->match_vlan_flags = vlan_flags;
+		pbr_map_check(pbrms, true);
```

Comment:
Same concern as above for PCP: How does the code distinguish between "match VLAN_ID==0" and "ignore VLAN_ID" ?

Reply:
In this case, VLAN_ID should only be a value between 1-4094 (see https://datatracker.ietf.org/doc/html/rfc5517). ID 0 is reserved, and handled by the untagged_0 match, so a value of zero is *always* ignored unless you do match `match untagged-or-zero`, rather than `match 0`. Eli documented this in the .rst and it is has special casing already in pbr_vty.c where it specifically excludes the value of 0 and matches instead on the constants defined in the pbr header (PBR_MAP_VLAN_UNTAGGED_0, PBR_MAP_VLAN_UNTAGGED, etc.). So, from my understanding these cases are distinct already.

Reply:
Thanks - it's a good explanation and I agree.